src/js/components/Select/index.d.ts

```diff
@@ -26,35 +44,43 @@ export interface SelectProps {
   dropTarget?: object;
   dropProps?: DropProps;
   focusIndicator?: boolean;
-  icon?: boolean | ((...args: any[]) => any) | React.ReactNode;
+  icon?:
+    | boolean
+    | ((props: {
+        color: ColorType;
+        size: 'small' | 'medium' | 'large' | 'xlarge' | string;
+      }) => React.ReactNode)
+    | React.ReactNode;
   id?: string;
-  labelKey?: string | ((...args: any[]) => any);
+  labelKey?: string | ((option: OptionType) => React.ReactNode);
   margin?: MarginType;
   messages?: { multiple?: string };
   multiple?: boolean;
   name?: string;
-  onChange?: (...args: any[]) => void;
-  onClose?: (...args: any[]) => any;
-  onMore?: (...args: any[]) => any;
-  onOpen?: (...args: any[]) => any;
+  onChange?: (event: onChangeEvent<OptionType, ValueType>) => void;
+  onClose?: () => void;
+  onMore?: () => void;
+  onOpen?: () => void;
   onSearch?: (search: string) => void;
-  options: (string | boolean | number | JSX.Element | object)[];
+  options: OptionType[];
   open?: boolean;
   placeholder?: PlaceHolderType;
   plain?: boolean;
   replace?: boolean;
   searchPlaceholder?: string;
-  selected?: number | number[];
+  selected?: number[] | number;
   size?: 'small' | 'medium' | 'large' | 'xlarge' | string;
-  value?: string | JSX.Element | object | (string | number | object)[];
+  value: ValueType;
```

Comment:
```suggestion
  value?: ValueType;
```
Value should have `?` to support uncontrolled forms (see Uncontrolled Select story)